modules/Multimode-Networks/src/main/java/cz/cvut/fit/gephi/multimode/MultiModeWindowTopComponent.java

```diff
@@ -0,0 +1,344 @@
+                    .addComponent(rightLabel))
+                .addGap(18, 18, 18)
+                .addComponent(removeEdges)
+                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                .addComponent(removeNodes)
+                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                .addComponent(start)
+                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                .addComponent(jSeparator1, javax.swing.GroupLayout.PREFERRED_SIZE, 10, javax.swing.GroupLayout.PREFERRED_SIZE)
+                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
+                    .addComponent(graphColoring)
+                    .addComponent(bipartiteLabel))
+                .addContainerGap(60, Short.MAX_VALUE))
+        );
+    }// </editor-fold>//GEN-END:initComponents
+
+    private void attributesActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_attributesActionPerformed
+
+        firstMatrix.removeAllItems();
+        secondMatrix.removeAllItems();
+        start.setEnabled(false);
+        if (attributes.getSelectedIndex() > 0) {
+            Column col = (Column) attributes.getSelectedItem();
+
+            // load all possible values of attribute
+            GraphController graphController = Lookup.getDefault().lookup(GraphController.class);
+            Graph graph = graphController.getGraphModel().getGraphVisible();
+            Node[] nodes = graph.getNodes().toArray();
+            Set<String> vals = new HashSet<String>();
+            for (Node n : nodes) {
+                Object val = n.getAttribute(col);
+                if (val != null) {
+                    vals.add(val.toString());
+                } else {
+                    vals.add("null");
+                }
+            }
+            // add combinations
+            for (String left : vals) {
+                for (String right : vals) {
+                    firstMatrix.addItem(new ValueCombination(left, right));
+                }
+            }
+            if (firstMatrix.getItemCount() > 0) {
+                firstMatrix.setSelectedIndex(0);
+            }
+        }
+    }//GEN-LAST:event_attributesActionPerformed
+
+    private void firstMatrixActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_firstMatrixActionPerformed
+        // TODO reaction on selected first matrix                
+        secondMatrix.removeAllItems();
+        if (attributes.getSelectedIndex() > 0) {
+
+            Column col = (Column) attributes.getSelectedItem();
+
+            // load all possible values of attribute
+            GraphController graphController = Lookup.getDefault().lookup(GraphController.class);
+            Graph graph = graphController.getGraphModel().getGraphVisible();
```

Comment:
Ideally one should check if there is any open project before accessing other controllers data such as GraphController.getGraphModel (here and in every place).

But the reality is many plugins don't do this so they will cause exceptions if no project is open.
So, if you can improve this, it would be great but we won't impose it in order to do the merge since it's not a critical problem.

This is done via ProjectController.
For example: Lookup.getDefault().lookup(ProjectController.class).getCurrentProject()
But in most cases it's useful to add a WorkspaceListener to keep track of projects opening and closing.

Real example at https://github.com/gephi/gephi/blob/master/modules/DesktopDataLaboratory/src/main/java/org/gephi/desktop/datalab/DataTableTopComponent.java#L310 from lines 310 to 347